rs/boundary_node/rate_limits/canister_client/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+use anyhow::{bail, Context, Error, Result};
+use candid::{Decode, Encode, Principal};
+use ic_agent::Agent;
+use rate_limits_api::{
+    v1::{RateLimitRule, SCHEMA_VERSION},
+    AddConfigError, AddConfigResponse, IncidentId, InputConfig, InputRule,
+};
+use serde::Deserialize;
+use std::{fs, path::PathBuf, str};
+use tracing::{debug, info};
+use tracing_subscriber::EnvFilter;
+use uuid::Uuid;
+
+pub async fn submit_config(
+    config_file: PathBuf,
+    canister_id: Principal,
+    agent: Agent,
+    debug: bool,
+) -> Result<(), Error> {
+    // initialize tracing subscriber with corresponding log level
+    let log_level = if debug { "debug" } else { "info" };
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::new(log_level))
+        .init();
+
+    // read the rules
+    let rules = read_yaml_file(&config_file);
```

Comment:
if `submit_config()` returns result, then ideally `read_yaml_file()` should also return result and not panic